src/query/service/src/pipelines/builders/builder_window.rs

```diff
@@ -169,6 +171,14 @@ impl PipelineBuilder {
             })
             .collect::<Result<Vec<_>>>()?;
 
+        if let Some(limit) = window_partition.limit {
+            if limit > 0 {
+                self.main_pipeline.add_transformer(|| {
+                    TransformWindowPartialTopN::new(partition_by.clone(), sort_desc.clone(), limit)
+                })
+            }
+        }
+
         self.main_pipeline.exchange(
             num_processors,
             WindowPartitionExchange::create(partition_by.clone(), num_partitions),
```

Comment:
I was thinking that combining TransformWindowPartialTopN and WindowPartitionExchange to form WindowPartitionTopNExchange should be a better implementation than the current one